com.ibm.wala.core/src/com/ibm/wala/ipa/cfg/AbstractInterproceduralCFG.java

```diff
@@ -401,34 +402,38 @@ public void removeNodeAndEdges(BasicBlockInContext<T> N) throws UnsupportedOpera
    */
   @Override
   public Iterator<BasicBlockInContext<T>> iterator() {
-    if (WARN_ON_EAGER_CONSTRUCTION) {
-      System.err.println("WARNING: forcing full ICFG construction by calling iterator()");
-    }
-    if (FAIL_ON_EAGER_CONSTRUCTION) {
-      throw new UnimplementedError();
-    }
-    constructFullGraph();
+    constructFullGraph("iterator");
     return g.iterator();
   }
 
+  /*
+   * @see com.ibm.wala.util.graph.NodeManager#iterateNodes()
+   */
+  @Override
+  public Stream<BasicBlockInContext<T>> stream() {
+    constructFullGraph("stream");
+    return g.stream();
+  }
+
   /*
    * @see com.ibm.wala.util.graph.NodeManager#getNumberOfNodes()
    */
   @Override
   public int getNumberOfNodes() {
+    constructFullGraph("getNumberOfNodes");
+    return g.getNumberOfNodes();
+  }
+
+  private boolean constructedFullGraph = false;
+
+  private void constructFullGraph(String onBehalfOf) {
     if (WARN_ON_EAGER_CONSTRUCTION) {
-      System.err.println("WARNING: forcing full ICFG construction by calling getNumberOfNodes()");
+      System.err.format("WARNING: forcing full ICFG construction by calling %s()\n", onBehalfOf);
```

Comment:
👍 Nice refactoring

Reply:
Thanks! I am particularly pleased with my choice of `onBehalfOf` as the formal parameter name. :nerd_face: